docs/data/data-grid/filtering/QuickFilteringDiacritics.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  GridToolbar,
+  GridColDef,
+  GridFilterModel,
+} from '@mui/x-data-grid';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Switch from '@mui/material/Switch';
+
+const rows = [
+  { id: 0, string: 'Café', date: new Date(2023, 1, 1), singleSelect: 'Jalapeño' },
+];
+const columns: GridColDef[] = [
+  { field: 'string', width: 100 },
+  {
+    field: 'date',
+    type: 'date',
+    width: 150,
+    valueFormatter: (params) =>
+      params.value.toLocaleDateString('fr-FR', {
+        day: 'numeric',
+        month: 'long',
+        year: 'numeric',
+      }),
```

Comment:
Sidenote but this is slow for formatters, not sure if we should show the more performant way in this example though. (using `Intl.DateTimeFormat` directly)

Reply:
Interesting, I didn't think about performance in this case.
However, the difference seems to be minimal (`toLocaleDateString` being a bit faster): https://jsperf.app/lohope

<img width="981" src="https://github.com/mui/mui-x/assets/13808724/57620258-c0fa-4353-b68f-3a2e8b599ee6">

Am I missing something?

Reply:
`.toLocaleDateString` builds a new `Intl.DateTimeFormat` instance on each call. It's best to build an instance outside the loop, and re-use it for each call.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString

Reply:
Another sidenot but I have also noticed in previous performance benchmarking that using the native `Intl` formatter vs using a JS based one is much slower. Crossing the JS/C++ boundary is usually quite bad for performance.

Reply:
Oh, gotcha!
Here's a benchmark for the singleton formatter: https://jsperf.app/hutiwe

<img width="980" src="https://github.com/mui/mui-x/assets/13808724/95925c4f-1012-4c69-85df-9d0a9b0f0439">

I'll update the demo.

> using the native Intl formatter vs using a JS based one is much slower

I didn't know that, thanks for the heads-up 👍🏻
I think for demo purposes `Intl.DateTimeFormat` is fine.